app/models/api/DistributedNodesRegistry.scala

```diff
@@ -0,0 +1,181 @@
+package models.api
+
+import akka.actor.{Actor, ActorLogging, ActorRef, ActorSystem, Props, TypedActor}
+import akka.cluster.Cluster
+import akka.cluster.ddata.{ORSet, ORSetKey}
+import akka.cluster.ddata.Replicator.Update
+import akka.cluster.ddata.Replicator._
+import akka.util.Timeout
+import com.google.inject.Inject
+import models.Settings.QueryTimeout
+import models.akka.StandardActions
+import models.rpc.DSAValue.StringValue
+import models.rpc.{InvokeRequest, RemoveRequest, SetRequest}
+
+import scala.annotation.tailrec
+import scala.concurrent.Future
+
+class DistributedNodesRegistry @Inject()(val replicator: ActorRef)(implicit cluster: Cluster, system: ActorSystem) extends Actor with ActorLogging {
```

Comment:
Do we need `cluster` as an implicit parameter? We can just take it from the `system`...